packages/wallets/src/hooks/useWalletAccountSwitcher.ts

```diff
@@ -0,0 +1,23 @@
+import { useCallback } from 'react';
+import { useAuthorize, useWalletAccountsList } from '@deriv/api-v2';
+
+const useWalletAccountSwitcher = () => {
+    const { switchAccount: _switchAccount } = useAuthorize();
+    const { data: walletAccounts } = useWalletAccountsList();
+
+    const switchAccount = useCallback(
+        (loginid: string) => {
+            const dtradeAccount = walletAccounts
+                ?.find(account => account.loginid === loginid)
+                ?.linked_to?.find(linkedAccount => linkedAccount.platform === 'dtrade');
+
+            _switchAccount(loginid);
+            if (dtradeAccount?.loginid) localStorage.setItem('active_loginid', dtradeAccount.loginid);
```

Comment:
ideally we shouldn't modify active_loginid outside of AuthProvider, it should just handle all the stuff properly